contracts/cw1-whitelist-ng/src/multitest/contract.rs

```diff
@@ -0,0 +1,239 @@
+        bail!("migrate not implemented for contract")
+    }
+}
+
+#[derive(PartialEq, Debug)]
+#[must_use]
+pub struct Cw1Executor<'a, App> {
+    addr: Addr,
+    app: &'a mut App,
+    sender: Addr,
+    send_funds: &'a [Coin],
+}
+
+impl<'a, App> Cw1Executor<'a, App> {
+    pub fn new(addr: Addr, app: &'a mut App, sender: Addr, send_funds: &'a [Coin]) -> Self {
+        Self {
+            addr,
+            app,
+            sender,
+            send_funds,
+        }
+    }
+
+    pub fn execute<C>(self, msgs: Vec<CosmosMsg<C>>) -> AnyResult<AppResponse>
+    where
+        C: Clone + std::fmt::Debug + PartialEq + JsonSchema + Serialize + 'static,
+        App: Executor<C>,
+    {
+        self.app.execute_contract(
+            self.sender,
+            self.addr,
+            &Cw1ExecMsg::Execute { msgs },
+            self.send_funds,
+        )
+    }
+}
+
+#[derive(PartialEq, Debug)]
+#[must_use]
+pub struct WhitelistExecutor<'a, App> {
+    addr: Addr,
+    app: &'a mut App,
+    sender: Addr,
+    send_funds: &'a [Coin],
+}
+
+impl<'a, App> WhitelistExecutor<'a, App> {
+    pub fn new(addr: Addr, app: &'a mut App, sender: Addr, send_funds: &'a [Coin]) -> Self {
+        Self {
+            addr,
+            app,
+            sender,
+            send_funds,
+        }
+    }
+
+    pub fn freeze<C>(self) -> AnyResult<AppResponse>
+    where
+        C: Clone + std::fmt::Debug + PartialEq + JsonSchema + Serialize + 'static,
+        App: Executor<C>,
+    {
+        self.app.execute_contract(
+            self.sender,
+            self.addr,
+            &WhitelistExecMsg::Freeze {},
+            self.send_funds,
+        )
+    }
+
+    pub fn update_admins<C>(self, admins: Vec<String>) -> AnyResult<AppResponse>
+    where
+        C: Clone + std::fmt::Debug + PartialEq + JsonSchema + 'static + Serialize,
+        App: Executor<C>,
+    {
+        self.app.execute_contract(
+            self.sender,
+            self.addr,
+            &WhitelistExecMsg::UpdateAdmins { admins },
+            self.send_funds,
+        )
+    }
+}
+
+#[derive(PartialEq, Debug)]
+#[must_use]
+pub struct Instantiator<'a, App> {
+    code_id: u64,
+    app: &'a mut App,
+    sender: Addr,
+    send_funds: &'a [Coin],
+    label: String,
+    admin: Option<String>,
+}
+
+impl<'a, App> Instantiator<'a, App> {
+    pub fn new(code_id: u64, app: &'a mut App, sender: Addr, send_funds: &'a [Coin]) -> Self {
+        Self {
+            code_id,
+            app,
+            sender,
+            send_funds,
+            label: "Cw1Whitelist".to_owned(),
+            admin: None,
+        }
+    }
+
+    pub fn with_label(mut self, label: &str) -> Self {
+        self.label = label.to_owned();
+        self
+    }
+
+    pub fn with_admin(mut self, admin: &str) -> Self {
+        self.admin = Some(admin.to_owned());
+        self
+    }
+
+    pub fn with<C>(self, admins: Vec<String>, mutable: bool) -> AnyResult<Cw1WhitelistProxy>
```

Comment:
I don't like this name. The other optional `with_{admin, label}` methods are great.

This should have a more different name. Maybe `do`? I'm sure you can come up with something better...

`let contract = Instantiator::new(....).with(...).unwrap();` looks a bit weird to me,.

Reply:
I don't like `do` to much. I see your point, I somehow agree. Maybe `call`? `perform`? `with_args`? I am struggling. Maybe someone have an idea?